pyscraper/new_hansard.py

```diff
@@ -0,0 +1,1442 @@
+        self.root = None
+        self.input_root = None
+        self.output_heading = False
+
+    def get_tag_name_no_ns(self, tag):
+        # remove annoying namespace for brevities sake
+        tag_name = str(tag.tag)
+        tag_name = tag_name.replace(
+            '{{{0}}}'.format(self.ns),
+            ''
+        )
+        return tag_name
+
+    def get_pid(self):
+        if self.current_speech_num == -1:
+            self.current_speech_num = 0
+        return '{0}{1}.{2}/{3}'.format(
+            self.rev,
+            self.current_col,
+            self.current_speech_num,
+            self.current_speech_part
+        )
+
+    def get_speech_id(self):
+        if self.current_speech_num == -1:
+            self.current_speech_num = 0
+        return 'uk.org.publicwhip/{0}/{1}{2}.{3}.{4}'.format(
+            self.debate_type,
+            self.date,
+            self.rev,
+            self.current_col,
+            self.current_speech_num
+        )
+
+    def check_for_pi(self, tag):
+        pi = tag.xpath('.//processing-instruction("notus-xml")')
+        if len(pi) == 1:
+            self.parse_pi(pi[0])
+
+    # this just makes any gid redirection easier
+    def get_text_from_element(self, el):
+        text = self.get_single_line_text_from_element(el)
+        text = u'\n{0}\n'.format(text)
+        return text
+
+    def get_single_line_text_from_element(self, el):
+        text = u''.join(el.xpath('.//text()'))
+        text = re.sub('\n', ' ', text).strip()
+        return text
+
+    def clear_current_speech(self):
+        if self.current_speech is not None:
+            self.root.append(self.current_speech)
+        self.current_speech_part = 1
+        if self.current_speech_num == -1:
+            self.current_speech_num = 0
+        else:
+            self.current_speech_num = self.current_speech_num + 1
+        self.current_speech = None
+
+    def new_speech(self, member, url):
+        if self.current_speech is not None:
+            self.clear_current_speech()
+        self.current_speech = etree.Element('speech')
+        self.current_speech.set('id', self.get_speech_id())
+        if member is not None:
+            self.current_speech.set('person_id', member['person_id'])
+            self.current_speech.set('speakername', member['name'])
+        else:
+            self.current_speech.set('nospeaker', 'true')
+        self.current_speech.set('colnum', self.current_col)
+        self.current_speech.set('time', self.current_time)
+        self.current_speech.set(
+            'url',
+            'http://www.publications.parliament.uk{0}'.format(
+                url
+            )
+        )
+
+    def parse_system_header(self, header):
+        sitting = header.xpath('./ns:Sitting', namespaces=self.ns_map)[0]
+        date = mx.DateTime.DateTimeFrom(sitting.get('short-date')).date
+        if date:
+            self.date = date
+
+    def handle_minus_member(self, member):
+        return None
```

Comment:
Comment please :) (It's for subclass overriding where that has a way of dealing with this entity.)